substrate/frame/sassafras/src/lib.rs

```diff
@@ -0,0 +1,1059 @@
+					continue
+				}
+
+				if TicketsData::<T>::contains_key(ticket_id) {
+					debug!(target: LOG_TARGET, "Ignoring duplicate ticket ({:032x})", ticket_id);
+					continue
+				}
+
+				let sign_data = vrf::ticket_body_sign_data(&ticket.body, ticket_id_input);
+
+				if ticket.signature.ring_vrf_verify(&sign_data, &verifier) {
+					TicketsData::<T>::set(ticket_id, Some(ticket.body));
+					valid_tickets
+						.try_push(ticket_id)
+						.expect("input segment has same length as bounded destination vector; qed");
+				} else {
+					debug!(target: LOG_TARGET, "Proof verification failure");
+				}
+			}
+
+			if !valid_tickets.is_empty() {
+				Self::append_tickets(valid_tickets);
+			}
+
+			Ok(Pays::No.into())
+		}
+
+		/// Plan an epoch config change.
+		///
+		/// The epoch config change is recorded and will be announced at the begin of the
+		/// next epoch together with next epoch authorities information.
+		/// In other words the configuration will be activated one epoch after.
+		/// Multiple calls to this method will replace any existing planned config change that had
+		/// not been enacted yet.
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::plan_config_change())]
+		pub fn plan_config_change(
+			origin: OriginFor<T>,
+			config: EpochConfiguration,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+
+			ensure!(
+				config.redundancy_factor != 0 && config.attempts_number != 0,
+				Error::<T>::InvalidConfiguration
+			);
+			PendingEpochConfigChange::<T>::put(config);
+			Ok(())
+		}
+	}
+
+	#[pallet::validate_unsigned]
+	impl<T: Config> ValidateUnsigned for Pallet<T> {
+		type Call = Call<T>;
+
+		fn validate_unsigned(source: TransactionSource, call: &Self::Call) -> TransactionValidity {
+			let Call::submit_tickets { tickets } = call else {
+				return InvalidTransaction::Call.into()
+			};
+
+			// Discard tickets not coming from the local node or that are not
+			// yet included in a block
+			debug!(
+				target: LOG_TARGET,
+				"Validating unsigned from {} source",
+				match source {
+					TransactionSource::Local => "local",
+					TransactionSource::InBlock => "in-block",
+					TransactionSource::External => "external",
+				}
+			);
+
+			if source == TransactionSource::External {
+				// TODO @davxy: BRAINSTORM this `Local` requirement...
+				// If we only allow these txs on block production, then there is less chance to
+				// submit our tickets if we don't have enough authoring slots.
+				// If we have 0 slots => we have zero chances.
+				// Maybe this is one valid reason to introduce proxies.
+				// In short the question is >>> WHO HAS THE RIGHT TO SUBMIT A TICKET? <<<
+				//  A) The current epoch validators
+				//  B) Doesn't matter as far as the tickets are good (i.e. RVRF verify is ok)
+				// Maybe we also provide a signed extrinsic to submit tickets
+				// where the submitter doesn't pay if the tickets are good?
```

Comment:
Yes! This would also allows us to make the submitter pay for useless tickets (duplicate, invalid, etc.), while still accepting the good tickets. Operationally it's more complicated though, since it requires that submitters (e.g. validators) have an account at hand with funds to pay for the fees (even though they will be reimbursed). This is the main reason we're using unsigned statements everywhere right now.

Reply:
I'm also thinking about mixnet relayers.

I don't know the details yet, but if a mixnet relay is allowed to not be a validator then this would be an application.
That is, allow any full node which participate in the mixnet network to submit tickets at any time (and not just IF they are validators  and WHEN is their turn to author a block).

The only requirement, as you said, is that the relays need to have an account with founds to pay for txs. 
But they will get reimbursed for any correct ticket they submit (the pay only for bad tickets...)

I leave this here as an element of discussion when I'll try to integrate with mixnet later (https://github.com/paritytech/polkadot-sdk/pull/1577#issuecomment-1812707909)